code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/utility/FileUtils.kt

```diff
@@ -0,0 +1,199 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.internal.utility
+
+import com.adobe.marketing.mobile.LoggingMode
+import com.adobe.marketing.mobile.MobileCore
+import java.io.BufferedReader
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.lang.Exception
+import java.lang.StringBuilder
+import java.util.zip.ZipInputStream
+
+object FileUtils {
+    const val TAG = "FileUtils"
+    private const val MAX_BUFFER_SIZE = 4096
+
+    /**
+     * Verifies if the directory is writable
+     *
+     * @param directory the directory to check for
+     * @return true if [directory] represents a directory and the directory is writable.
+     *         false otherwise
+     */
+    @JvmStatic
+    fun isWritableDirectory(directory: File?): Boolean {
```

Comment:
This is the utility class only used inside the Core project, can we make all `File` parameters non-optional and add a `@NonNull` annotation for them?

Reply:
I actually prefer if this class acts as a true utility class that reduces the null check boilerplate code on the caller. That said, I will change it if you feel strongly about making them non null.

Reply:
Not having a strong feeling about the above recommendation. We can leave it as is if the callers have to do null checks everywhere.